Extensions/PlatformBehavior/platformerobjectruntimebehavior.ts

```diff
@@ -1251,11 +1089,458 @@ namespace gdjs {
     isMoving(): boolean {
       return (
         (this._hasReallyMoved && this._currentSpeed !== 0) ||
-        this._currentJumpSpeed !== 0 ||
+        this._jumping.getCurrentJumpSpeed() !== 0 ||
         this._currentFallSpeed !== 0
       );
     }
   }
+
+  export namespace PlatformerObjectRuntimeBehavior {
```

Comment:
Is it possible to avoid exporting these interface/classes? There are internal logic and so should not be visible to the outside if possible.

Reply:
I tried to write a private inner classes with this:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/class
but it would require to declare them before the behavior attributes declaration and it looks a bit messy.

Reply:
Sorry I was not clear: let's not put the classes inside the behavior as member, let's just declare them without an `export` keyword and it should work.
If we must move them before the declaring of the behavior class that's fine - as long as 1) it's not inside the behavior class and 2) they are not `export`ed, i.e: you can't do `gdjs.PlatformerObjectRuntimeBehavior.OnFloor` from the outside. It's fine if you refer as them as `OnFloor` inside the extension itself, but if you have to use something like `this.OnFloor`, it's a no go as it means you'll have one class per instance, which will be a nightmare for JS engines :)